src/main/java/io/github/moulberry/notenoughupdates/miscfeatures/GardenNpcPrices.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2023 NotEnoughUpdates contributors
+ *
+ * This file is part of NotEnoughUpdates.
+ *
+ * NotEnoughUpdates is free software: you can redistribute it
+ * and/or modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation, either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * NotEnoughUpdates is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with NotEnoughUpdates. If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.moulberry.notenoughupdates.miscfeatures;
+
+import io.github.moulberry.notenoughupdates.NotEnoughUpdates;
+import io.github.moulberry.notenoughupdates.autosubscribe.NEUAutoSubscribe;
+import io.github.moulberry.notenoughupdates.util.ItemResolutionQuery;
+import io.github.moulberry.notenoughupdates.util.Utils;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.Item;
+import net.minecraftforge.event.entity.player.ItemTooltipEvent;
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+@NEUAutoSubscribe
+public class GardenNpcPrices {
+
+	private final Pattern itemRegex = Pattern.compile("§5§o §.([a-zA-Z \\-]+)(?:§8x(\\d+))?");
+	//§5§o §aEnchanted Cactus Green §8x421
+	//§5§o §aEnchanted Hay Bale §8x62
+	//§5§o §9Enchanted Cookie §8x4
+	//§5§o §9Tightly-Tied Hay Bale
+
+	private HashMap<Integer, List<String>> prices = new HashMap<>();
```

Comment:
Also: this cache never gets cleared (neither on `RepositoryReloadEvent`) neither on world swap nor kept below a size (Maybe use LRUCache?)

Reply:
> Also why do you use the hashcode as key instead of the actual tooltip (or a copy of that)? Javas hashcodes are not meant to be collision free

idk memory ig?

Reply:
> idk memory ig?
bruh. how long are your tooltips

Reply:
💀 also ik im probably being stupid but just putting the list in doesn't work because they are technically different lists

Reply:
they should still be compared using .equals() (which by contract is just a stricter version of comparing .hashcode()) my guess is that the tooltip array is being modified by later event handlers (so we would need to store a copy)